internal/plugin/plugin_executable.go

```diff
@@ -0,0 +1,79 @@
+package plugin
+
+import (
+	"github.com/hashicorp/boundary/internal/plugin/store"
+)
+
+// OperatingSystem defines the operating systems supported for plugin executables
+type OperatingSystem string
+
+const (
+	UnknownOS   OperatingSystem = "unknown"
```

Comment:
Again, might want to restrict this to just the arches we actually support for now.